Dev 3.8

GigaZiv opened the pull request:
Выполнено задание на 15 спринт

С уважением Жуков Игорь

GigaZiv committed:
Dev 3.8
Выполнено задание на 15 спринт

С уважением Жуков Игорь

ArturNurtdinov commented on app/src/main/java/rs/example/playlistmaker/data/dto/CustomException.kt:
```diff
@@ -0,0 +1,3 @@
+package rs.example.playlistmaker.data.dto
+
+class CustomException(message: String?) : Exception()
```
Такой класс можно не создавать, а просто создавать экземпляры класса `Exception`, но лучше все-таки более понятные ошибки выбрасывать, например `IllegalStateException`

ArturNurtdinov commented on app/src/main/java/rs/example/playlistmaker/domain/impl/MediaPlayerControlIpm.kt:
```diff
@@ -0,0 +1,89 @@
+package rs.example.playlistmaker.domain.impl
+
+import android.media.MediaPlayer
+import android.util.Log
```
В домейн слое не должно быть зависимостей от Android SDK - это будет в след спринте, поэтому сразу тут отмечу, что работу с логгером и медиаплеером нужно будет вынести в data

ArturNurtdinov commented on app/src/main/java/rs/example/playlistmaker/domain/impl/MediaPlayerControlIpm.kt:
```diff
@@ -0,0 +1,89 @@
+import rs.example.playlistmaker.utils.StaffFunctions.getSimpleDateFormatInt
+
+
+class MediaPlayerControlIpm(val mediaInterface: MediaPlayerControlInterface) {
```
От неймингов Controller, Helper и других лучше уходить в сторону Interactor и Repository

GigaZiv committed:
Dev 3.8.1
У меня запускается и компилируется. Переименовал файл SearchTrackInstance

С уважением Жуков Игорь

GigaZiv committed:
Dev 3.8.2

Удалил обработчик Custom

С уважением Жуков Игорь

GigaZiv committed:
Dev 3.8.3
Все добавил....

С уважением Жуков Игорь
